hs/src/Reach/Simulator/Core.hs

```diff
@@ -631,28 +634,63 @@ instance Interp LLStep where
       _ <- interp stmt
       interp step
     LLS_Stop _at -> return V_Null
-    LLS_ToConsensus _at _lct tc_send (DLRecv {..}) _tc_mtime -> do
+    LLS_ToConsensus _at _lct tc_send recv tc_mtime -> do
       (g, l) <- getState
       let mid = e_nmsgid g
       incrMessageId
-      v <- suspend $ PS_Suspend (A_TieBreak mid $ M.keys $ M.mapKeys bunpack tc_send)
-      case v of
-        V_UInt n -> do
-          let locals = e_locals l
-          let lclsv = saferMapRef "LLS_ToConsensus" $ M.lookup (fromIntegral n) locals
-          let lclst = l_who $ lclsv
-          case lclst of
-            Nothing -> impossible "expected participant id, received consensus id"
-            Just part -> do
-              let m = saferMapRef "LLS_ToConsensus1" $ M.lookup (bpack part) tc_send
-              case m of
-                DLSend {..} -> do
+      let sends = M.mapKeys bunpack tc_send
+      v <- suspend $ PS_Suspend (A_TieBreak mid $ M.keys sends)
+      case tc_mtime of
+        Just (dltimearg, step) -> case dltimearg of
+          Left dlarg -> do
+            ev <- interp dlarg
```

Comment:
I think you need some helper functions like `vUInt` that throws the "unexpected" error to keep this code nicer: `vUInt <$> interp dlarg`